In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import re
%matplotlib inline
plt.style.use('ggplot')

In [5]:
!ls data

national_footprint_data.csv


In [7]:
footprint_df = pd.read_csv('data/national_footprint_data.csv')

In [12]:
footprint_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 87020 entries, 0 to 87019
Data columns (total 15 columns):
country                     87020 non-null object
ISO alpha-3 code            85660 non-null object
UN_region                   87020 non-null object
UN_subregion                87020 non-null object
year                        87020 non-null int64
record                      87020 non-null object
crop_land                   63530 non-null float64
grazing_land                63530 non-null float64
forest_land                 63530 non-null float64
fishing_ground              63530 non-null float64
built_up_land               63530 non-null float64
carbon                      63530 non-null float64
total                       87020 non-null float64
Percapita GDP (2010 USD)    71410 non-null float64
population                  87020 non-null int64
dtypes: float64(8), int64(2), object(5)
memory usage: 10.0+ MB


In [11]:
footprint_df['country'].unique().size

196

In [15]:
footprint_df['year'].unique().size

54

Our dataset spans 54 years and included 196 total countries

## Business Understanding 1 -- 10
* Describe the purpose of the data set you selected (i.e., why was this data collected in the first place?). How will you measure the effectiveness of a good algorithm? Why does your chosen validation method make sense for this specific dataset and the stakeholders needs?

## Data Understanding 1 --10
* Describe the meaning and type of data (scale, values, etc.) for each attribute in the data file. Verify data quality: Are there missing values? Duplicate data? Outliers? Are those mistakes? How do you deal with these problems?

## Data Understanding 2 -- 	10
* Visualize the any important attributes appropriately. Important: Provide an interpretation for any charts or graphs.

## Modeling and Evaluation 1 -- 10
* Train and adjust parameters

## Modeling and Evaluation 2 -- 10 
* Evaluate and Compare

## Modeling and Evaluation 3 -- 10	
* Visualize Results

## Modeling and Evaluation 4 -- 20
* Summarize the Ramifications

## Deployment 	10	
* Be critical of your performance and tell the reader how you current model might be usable by other parties. Did you achieve your goals? If not, can you reign in the utility of your modeling? How useful is your model for interested parties (i.e., the companies or organizations that might want to use it)? How would your deploy your model for interested parties? What other data should be collected? How often would the model need to be updated, etc.?

## Exceptional Work	10	
* You have free reign to provide additional analyses or combine analyses.